src/Take.Elephant.Sql/ExpirableKeySqlMap.cs

```diff
@@ -157,4 +161,69 @@ private static string InjectSqlFilter(string sql, string filter)
             }
         }
     }
+
+    /// <summary>
+    /// Implements a new behavior for DbConnection with a custom parameter.
+    /// </summary>
+    internal class ExpirationDbConnectionDecorator : DbConnection
+    {
+        private readonly DbConnection _dbConnection;
+        public ExpirationDbConnectionDecorator(DbConnection dbConnection)
+        {
+            _dbConnection = dbConnection;
+        }
+
+        public override string ConnectionString { get => _dbConnection.ConnectionString; set => _dbConnection.ConnectionString = value; }
+
+        public override string Database => _dbConnection.Database;
+
+        public override string DataSource => _dbConnection.DataSource;
+
+        public override string ServerVersion => _dbConnection.ServerVersion;
+
+        public override ConnectionState State => _dbConnection.State;
+
+        public override void ChangeDatabase(string databaseName)
+        {
+            _dbConnection.ChangeDatabase(databaseName);
+        }
+
+        public override void Close()
+        {
+            _dbConnection.Close();
+        }
+
+        public override void Open()
+        {
+            _dbConnection.Open();
+        }
+
+        public override async ValueTask DisposeAsync()
+        {
+            await base.DisposeAsync();
+            await _dbConnection.DisposeAsync();
+        }
+
+        protected override DbCommand CreateDbCommand()
+        {
+            var command = _dbConnection.CreateCommand();
+            var parameter = command.CreateParameter();
+            parameter.ParameterName = ExpirableKeySqlMap<int,int>.ExpirationDatabaseDriverDecorator.EXPIRATION_DATE_PARAMETER_NAME;
```

Comment:
Putting this class inside ExpirabkeKeySqlMap - as ExpirationDatabaseDriverDecorator is - will make more clear that there are all related, and allow to remove the `ExpirableKeySqlMap<int,int>` here.

Reply:
But if I do it , I'll have to use `ExpirableKeySqlMap<int,int>` on test class.

Reply:
Yes, but I think it's better there than here...

Reply:
Ok, done